### ISSUE TITLE
New linter: detect duplicate cond tests

### ISSUE THREAD

**Issue by marco-carvalho** [clj-kondo/clj-kondo#2962]
- [ ] I have read the [Clojure etiquette](https://clojure.org/community/etiquette) and will respect it when communicating on this platform.

**Is your feature request related to a problem? Please describe.**

Repeating the same non-constant test in a `cond` makes the later branch unreachable and often indicates a copy and paste mistake.

```clojure
(cond
  (odd? x) :odd
  (odd? x) :also-odd)
```

**Describe the solution you'd like**

Add a `:duplicate-cond-test` linter, enabled at warning level by default, that reports the second and later structurally equal non-constant tests. Constant tests remain handled by existing condition linters. Generated forms should not warn.

**Describe alternatives you've considered**

This could be folded into `:unreachable-code`, but a dedicated linter gives a clearer diagnosis and can be configured independently. The open `cond-to-if` and `cond-as-case` proposals detect different patterns.

**Additional context**

I am interested in submitting a focused PR with tests and documentation if this approach is accepted.

**Comment by NoahTheDuke** [comment]
Reminds me of [`:case-duplicate-test`](https://github.com/clj-kondo/clj-kondo/blob/18448daa0ca2b53b2dddce5773f641bed3b5fc85/doc/linters.md#case-duplicate-test). Should that be generalized to cover `cond` as well?

**Comment by marco-carvalho** [comment]
Good point. They detect the same class of mistake. I initially kept this separate because `:case-duplicate-test` is named and documented specifically around case constants, while this check compares non-constant cond forms. Generalizing the existing concept is cleaner than adding another default-on linter, though the existing keyword makes that awkward. A backwards-compatible option would be to introduce a general keyword and retain `:case-duplicate-test` as an alias. I can revise the proposal in that direction if that migration shape is acceptable.
